src/dra/5087C.c

```diff
@@ -524,13 +524,41 @@ void func_800F2014(void) {
         if (!(currMapRect & subMap)) {
             g_CastleMap[idx] = currMapRect | subMap;
             g_RoomCount++;
-            func_800F1B08(x, y, 0, currMapRect);
+            func_800F1B08(x, y, 0);
             func_800F1EB0(x, y, 0xFFFF);
         }
     }
 }
 
-INCLUDE_ASM("dra/nonmatchings/5087C", func_800F2120);
+void func_800F2120(void) {
+    s32 x;
+    s32 y;
+    s32 subMap;
+    s32 idx;
+    u8 currMapRect;
+
+    func_800F1A3C(g_StageId & STAGE_INVERTEDCASTLE_FLAG);
+    ClearImage(&g_Vram.D_800ACDE8, 0, 0, 0);
+    DrawSync(0);
+
+    for (y = 0; y < 64; y++) {
+        for (x = 0; x < 64; x++) {
+            idx = (x >> 2) + (y * 16);
+            subMap = 3 << ((3 - (x & 3)) * 2);
```

Comment:
Commenting again to make this more clear:
```
            // sequence of 2 bit masks: 0xC0, 0x30, 0x0C, 0x03
            // 0b11000000, 0b110000, 0b1100, 0b11
 ```
 Could you add those two comments above `subMap = 3 << ((3 - (x & 3)) * 2);` to document what it does?

Reply:
Done, thank you for the clarification